app/controllers/admin/products_controller.rb

```diff
@@ -1,7 +1,60 @@
 # frozen_string_literal: true
 
 class Admin::ProductsController < Admin::AdminController
+  before_action :load_product, only: %i(update edit destroy)
+
   def index
     @pagy, @products = pagy(Product.get_all_by_name_sort, items: Settings.DIGIT_5)
   end
+
+  def edit
+    @hidden_partial = true
+  end
+
+  def new
+    @product = Product.new
+    @hidden_partial = true
+  end
+
+  def update
+    flash[:admin_success] = if @product.update(product_params)
+                              t("admin.products.update.success")
+                            else
+                              t("admin.products.update.fail")
+                            end
+    redirect_to(admin_products_path)
+  end
+
+  def create
+    @product = Product.new(product_params)
+    flash[:admin_success] = if @product.save
+                              t("admin.products.create.success")
+                            else
+                              t("admin.products.create.fail")
+                            end
+    redirect_to(admin_products_path)
+  end
+
+  def destroy
```

Comment:
Xóa product mà dễ như ri thì mấy cửa hàng sập tiệm hết

Reply:
Chỗ này em bổ sung thêm logic những sản phẩm nào đã được đặt hàng thì không được phép xóa, in ra thông báo cho người dùng